src/ussd_clinic_rapidpro.js

```diff
@@ -140,16 +140,77 @@ go.app = function() {
                         );
                     }
                 },
-                next: "state_get_contact"
+                next: "state_whatsapp_check"
             });
         });
 
+        self.add("state_whatsapp_contact_check", function(name, opts) {
+            var msisdn = utils.normalize_msisdn(
+                _.get(self.im.user.answers, "state_enter_msisdn", self.im.user.addr), "ZA");
+            self.whatsapp.contact_check(msisdn, false).then(_.noop, _.noop);
```

Comment:
If we're immediately doing a wait contact check after this, then this nowait check is not necessary. This is only an advantage if we can do it earlier on, so that when we do the wait check it returns quicker.